src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.h

```diff
@@ -10,12 +10,33 @@
 namespace DB
 {
 
+/** Stores data in immutable files, but allows atomic directory renames, which is suitable for MergeTree tables.
```

Comment:
I'm proud of every comment in my code, and I think everyone else must write comments.